pom.xml

```diff
@@ -662,15 +760,15 @@
                 </configuration>
             </plugin>
             <plugin>
+                <!-- Warning: do not put this plugin's configuration before surefire plugin one. The sequence
+                     in which the surefire and jacoco goals run is important -->
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
                 <version>${jacoco.version}</version>
                 <executions>
                     <execution>
                         <id>prepare-agent</id>
-                        <goals>
-                            <goal>prepare-agent</goal>
-                        </goals>
+                        <!-- Do not bind any goals, let a profile to do it conditionally -->
```

Comment:
The configuration that follows the lines you are removing here is configuration for the goal in question.  I don't think this works in general.

Reply:
Apologies I do not understand the comment:  The configuration you said is being removed is the only place in `pom.xml` that contains `-unit.exec` string.

Still, with this removal in place, the command line command `mvn -Dtest=XNetReplyTest test` prints out:
```
[DEBUG] === PROJECT BUILD PLAN ================================================
[DEBUG] -----------------------------------------------------------------------
[DEBUG] Goal:          org.jacoco:jacoco-maven-plugin:0.8.5:prepare-agent (prepare-agent)
[DEBUG] Style:         Regular
[DEBUG] Configuration: <?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <address>${jacoco.address}</address>
  <append>true</append>
  <classDumpDir>${jacoco.classDumpDir}</classDumpDir>
  <destFile default-value="${project.build.directory}/jacoco.exec">/space/src/jmri/fixes2/JMRI/target//jacoco-unit.exec</destFile>
  <dumpOnExit>${jacoco.dumpOnExit}</dumpOnExit>
  <exclClassLoaders>${jacoco.exclClassLoaders}</exclClassLoaders>
  <inclBootstrapClasses>${jacoco.inclBootstrapClasses}</inclBootstrapClasses>
  <inclNoLocationClasses>${jacoco.inclNoLocationClasses}</inclNoLocationClasses>
  <jmx>${jacoco.jmx}</jmx>
  <output>${jacoco.output}</output>
  <pluginArtifactMap>${plugin.artifactMap}</pluginArtifactMap>
  <port>${jacoco.port}</port>
  <project>${project}</project>
  <propertyName>surefire.argLine</propertyName>
  <sessionId>${jacoco.sessionId}</sessionId>
  <skip default-value="false">${jacoco.skip}</skip>
</configuration>

```
The configuration is IMHO not a property of a goal, but property of the execution instance with the given ID. The goal is "only" bound to that execution ID.

Later, the execution says:
```
[DEBUG] Configuring mojo 'org.jacoco:jacoco-maven-plugin:0.8.5:prepare-agent' with basic configurator -->
[DEBUG]   (f) append = true
[DEBUG]   (f) destFile = /space/src/jmri/fixes2/JMRI/target/jacoco-unit.exec
```

Maybe I didn't understood what you said was removed or what bad effect that removal has ?

Reply:
As long as this is working, I am fine with it.  The change I requested was to have the default-cli configuration restored for merge

Reply:
> As long as this is working, I am fine with it. The change I requested was to have the default-cli configuration restored for merge

Still I'd like to understand what did you see removed:
> The configuration that follows the lines you are removing here is configuration for the goal in question. I don't think this works in general.